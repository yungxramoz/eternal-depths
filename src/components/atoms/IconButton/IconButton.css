.icon-button {
  background-color: transparent;
  border: none;
  cursor: pointer;
  width: 4rem;
}