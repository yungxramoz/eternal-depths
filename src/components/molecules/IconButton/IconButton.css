.icon-button {
  color: #d27d2c;
  background-color: transparent;
  border: none;
  cursor: pointer;
  width: 4rem;
}

.icon-button:disabled {
  color: #999;
}

@media (hover: hover) {
  .icon-button:hover {
    filter: drop-shadow(-1px 0 #d27d2c) drop-shadow(1px 0 #d27d2c)
      drop-shadow(0 -1px #d27d2c) drop-shadow(0 1px #d27d2c);
  }
  /* disabled */
  .icon-button:disabled:hover {
    filter: drop-shadow(-1px 0 #999) drop-shadow(1px 0 #999)
      drop-shadow(0 -1px #999) drop-shadow(0 1px #999);
  }
}

.icon-button:active {
  color: #854c30;
}

.icon-button:disabled:active {
  color: #4e4a4e;
}

.icon-button.small {
  width: 1.5rem;
  height: 1.5rem;
}

.icon-button.medium {
  width: 2rem;
  height: 2rem;
}

.icon-button.large {
  width: 3rem;
  height: 3rem;
}
