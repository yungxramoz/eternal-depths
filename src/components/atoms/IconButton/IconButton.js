import React from 'react'
import './IconButton.css'

const IconButton = ({ iconName, onClick }) => {
  const iconPath = `./assets/images/ui/${iconName}.svg`

  return (
    <button className="icon-button" onClick={onClick}>
      <img src={iconPath} alt={iconName} />
    </button>
  )
}

export default IconButton
