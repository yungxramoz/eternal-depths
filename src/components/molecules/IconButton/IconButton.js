import React from 'react'
import ArrowLeftIcon from '../../atoms/Icons/ArrowLeftIcon'
import ChevronLeftIcon from '../../atoms/Icons/ChevronLeftIcon'
import ChevronRightIcon from '../../atoms/Icons/ChevronRightIcon'
import MinusIcon from '../../atoms/Icons/MinusIcon'
import PlusIcon from '../../atoms/Icons/PlusIcon'
import './IconButton.css'

const IconButton = ({ icon, size, disabled, className, onClick }) => {
  return (
    <button
      className={`icon-button ${getSizeClass(size)} ${className}`}
      disabled={disabled}
      onClick={onClick}
    >
      {getIcon(icon)}
    </button>
  )
}

const getSizeClass = (size) => {
  switch (size) {
    case 'small':
      return 'small'
    case 'medium':
      return 'medium'
    case 'large':
      return 'large'
    default:
      return ''
  }
}

const getIcon = (icon) => {
  switch (icon) {
    case 'chevron-left':
      return <ChevronLeftIcon />
    case 'chevron-right':
      return <ChevronRightIcon />
    case 'arrow-left':
      return <ArrowLeftIcon />
    case 'minus':
      return <MinusIcon />
    case 'plus':
      return <PlusIcon />
    default:
      return null
  }
}

export default IconButton
