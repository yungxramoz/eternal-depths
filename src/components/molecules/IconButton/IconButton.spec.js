import { render, screen } from '@testing-library/react'
import React from 'react'
import IconButton from './IconButton'

describe('IconButton', () => {
  it('renders IconButton component', () => {
    render(<IconButton icon="plus" />)
    expect(document.body).toMatchSnapshot('IconButton')
  })

  it('renders the correct size class based on the size prop', () => {
    render(<IconButton icon="plus" size="large" />)
    expect(document.body).toMatchSnapshot('IconButton Large')
  })

  it('disables the button when the disabled prop is true', () => {
    render(<IconButton icon="plus" disabled />)
    expect(document.body).toMatchSnapshot('IconButton Disabled')
  })
})
