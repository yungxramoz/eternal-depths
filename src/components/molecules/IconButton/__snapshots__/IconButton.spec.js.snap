// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton disables the button when the disabled prop is true: IconButton Disabled 1`] = `
<body>
  <div>
    <button
      class="icon-button  "
      disabled=""
    >
      <svg
        fill="currentColor"
        id="Plus"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          points="23 11 23 13 13 13 13 23 11 23 11 13 1 13 1 11 11 11 11 1 13 1 13 11 23 11"
        />
      </svg>
    </button>
  </div>
</body>
`;

exports[`IconButton renders IconButton component: IconButton 1`] = `
<body>
  <div>
    <button
      class="icon-button  "
    >
      <svg
        fill="currentColor"
        id="Plus"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          points="23 11 23 13 13 13 13 23 11 23 11 13 1 13 1 11 11 11 11 1 13 1 13 11 23 11"
        />
      </svg>
    </button>
  </div>
</body>
`;

exports[`IconButton renders the correct size class based on the size prop: IconButton Large 1`] = `
<body>
  <div>
    <button
      class="icon-button large "
    >
      <svg
        fill="currentColor"
        id="Plus"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          points="23 11 23 13 13 13 13 23 11 23 11 13 1 13 1 11 11 11 11 1 13 1 13 11 23 11"
        />
      </svg>
    </button>
  </div>
</body>
`;
